src/backend/access/common/reloptions_gp.c

```diff
@@ -1657,6 +1666,111 @@ List* transformColumnEncoding(Relation rel, List *colDefs, List *stenc, List *wi
 		 * 3. A default column encoding in the statement
 		 * 4. A default for the type.
 		 */
+		if (d->encoding)
+		{
+			if (errorOnEncodingClause)
+				ereport(ERROR,
+						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("ENCODING clause only supported with column oriented tables")));
+
+			encoding = tam->transform_column_encoding_clauses(rel, d->encoding, true, false);
+		}
+		else
+		{
+			ColumnReferenceStorageDirective *s = find_crsd(d->colname, stenc);
+
+			if (s) {
+				Assert(tam);
+				encoding = tam->transform_column_encoding_clauses(rel, s->encoding, true, false);
+			} else {
+				if (deflt)
+					encoding = copyObject(deflt->encoding);
+				else
+				{
+					if (d->typeName) {
+						/* get encoding by type, still need do transform and validate */
+						encoding = get_type_encoding(d->typeName);
+						if (tam && tam->transform_column_encoding_clauses)
+							encoding = tam->transform_column_encoding_clauses(rel, encoding, true, true);
+					}
```

Comment:
Do we need to check transform_column_encoding_clauses and transform_column_encoding_clauses is NULL here?

Reply:
It's hard to tell.